project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java

```diff
@@ -2071,6 +2078,45 @@ public void testBuildProjectWithUncompletedPackageInformation(String projectDir,
         Files.deleteIfExists(projectPath.resolve(DEPENDENCIES_TOML));
     }
 
+    @Test(description = "test accessing semantic model after first build",
+            dependsOnMethods = "testGetSemanticModel")
+    public void testAccessSemanticModelAfterFirstBuild() throws IOException {
+        Path projectPath = RESOURCE_DIRECTORY.resolve("myproject");
+        // Delete build file if already exists
+        if (projectPath.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE).toFile().exists()) {
+            Files.delete(projectPath.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE));
+        }
+        // Set sticky false, to imitate the default build command behavior
+        BuildOptions.BuildOptionsBuilder buildOptionsBuilder = BuildOptions.builder();
+        buildOptionsBuilder.setSticky(false);
+
+        // 1) Initialize the project instance
+        BuildProject project = loadBuildProject(projectPath, buildOptionsBuilder.build());
+        // Get compilation
+        PackageCompilation compilation = project.currentPackage().getCompilation();
+        compilation.diagnosticResult().diagnostics().forEach(OUT::println);
+        Assert.assertFalse(compilation.diagnosticResult().hasErrors());
+        // Call `JBallerinaBackend`
+        JBallerinaBackend.from(compilation, JvmTarget.JAVA_11);
+        // Check bir is generated for default module
+        Assert.assertTrue(project.targetDir().resolve(CACHES_DIR_NAME).resolve("sameera").resolve("myproject")
+                .resolve("0.1.0").resolve(REPO_BIR_CACHE_NAME).resolve("myproject.bir").toFile().exists());
+
+        // 2) Build project again with build file
+        // Set temp dir as the target
+        buildOptionsBuilder.targetDir(
+                String.valueOf(Files.createTempDirectory("ballerina-cache" + System.nanoTime())));
```

Comment:
Refer to the previous comment: https://github.com/ballerina-platform/ballerina-lang/pull/36737/files#r909552639

Reply:
Added